app/javascript/lib/scp-api.js

```diff
@@ -592,6 +594,15 @@ export default async function scpApi(
     } else {
       return [response, perfTimes, true]
     }
+  } else {
+    if (response.status === 401 || response.status === 403) {
+      showMessage(
+        <div>
+          Authentication failed<br/>
+          Your session may have timed out. Please sign in again<br/><br/>
+        </div>, 'api-auth-failure'
```

Comment:
Let's instrument this in a way that enables more analytics use cases, e.g.:

* Find all auth errors thrown by Rails non-API endpoints (currently logged via `error-modal`) and Rails API endpoints (logged via `message-modal`)
* Find all API errors, e.g. to make a graph of different errors
* Find errors by HTTP status code.  Requires less familiarity with our app-specific keys, useful if quickly comparing errors logged in Mixpanel to errors logged in e.g. Sentry or Stackdriver.

Likely not exactly this, but maybe something along the lines of:
```suggestion
        </div>, 'api-auth-failure', {
          source: 'api', // Did the server error come from the API controllers, other controllers, or elsewhere?
          url, // This has proven very useful in other Mixpanel use cases
          isOk: false, // Applies to both server and client errors
          messageType: 'client-error', // Applies for all 400-status errors
          statusCode: response.status // The specific standard, invariant error code
        }
```